firmware/lib/Kinematics/src/Kinematics.h

```diff
@@ -0,0 +1,45 @@
+#ifndef KINEMATICS_H
+#define KINEMATICS_H
+
+#include "Arduino.h"
+#include "Timer.h"
+#include "Encoder.h"
+#include "LimitSwitch.h"
+
+#define VELOCITY_HOMING 200
+
+enum status{pressed, depressed};
+enum direction{positive, negative};
+enum segment{accelerate, hold, decelerate};
+
+typedef struct Axis
+{
+    int dir_pin;
+    int step_pin;
```

Comment:
Any reason these are `int`s rather than `uint32_t`s. `pinMode` and `digitalWrite` both take `uint32_t` as an argument